test/testcommon/src/main/java/io/pravega/test/common/AssertExtensions.java

```diff
@@ -312,6 +312,35 @@ public static void assertNotNullOrEmpty(String message, String string) {
         Assert.assertFalse(message, string == null || string.length() == 0);
     }
 
+    /**
+     * Asserts that a future (that has not yet been invoked) throws an expected exception or completes without
+     * exception.
+     *
+     * @param message        The message to include in the Assert calls.
+     * @param futureSupplier A Supplier that returns a new CompletableFuture, to test.
+     * @param tester         A predicate that indicates whether the exception (if thrown) is as expected.
+     * @param <T>            The type of the future's result.
+     *
+     * @return boolean       Returns true if an expected exception is actually thrown.
+     */
+    public static <T> boolean assertMayThrow(String message, Supplier<CompletableFuture<T>> futureSupplier,
```

Comment:
I think I see what you are trying to do here, but I don't really like how this is defined. Please return void and remove the return statements.